### ISSUE TITLE
[iOS] No callback when user dismisses the login popup

### ISSUE THREAD

**Issue by kaushikpaperboat** [auth0/Auth0.swift#444]
### Describe the problem

So in the app we are calling following method to initiate the login flow (Swift):

```
Auth0
    .webAuth(clientId: {clientId}, domain: {domain})
    .scope("openid email profile")
    .parameters(["prompt":"select_account"])
    .connection({apple/facebook/google-oauth2})
    .useEphemeralSession()
    .logging(enabled: true)
    .start {}
```
Note that we are passing the **connection** parameter explicitly so we are essentially not using Universal Login provided by Auth0. On the login page on iPad user can dismiss the view by tapping anywhere outside the view area and we are relying on the callback to enable UI and show some messages. 

### What was the expected behavior?

Expected behaviour would be to receive cancel callback if tap anywhere outside the view.

### Reproduction

It can be reproduced in the sample app also, you just have to add `.connection({apple/facebook/google-oauth2})` call and there will be no callback if user taps outside the login view.

**Can the behavior be reproduced using the [Auth0.swift sample app](https://github.com/auth0-samples/auth0-ios-swift-sample)?**

Yes

### Environment

> Please provide the following:

- 1.30.1
- 14.3
- 12.4

Is this the expected behaviour or is it a bug?

**Comment by Widcket** [comment]
Hi @kaushikpaperboat, this seems to be an iPadOS bug. I managed to reproduce it without the SDK, just with `ASWebAuthenticationSession`.

```swift
import UIKit
import AuthenticationServices

class ViewController: UIViewController {
    var webAuthSession: ASWebAuthenticationSession?

    let authURL = URL(string: "https://YOUR_AUTH0_DOMAIN.auth0.com/authorize?nonce=1234&client_id=YOUR_AUTH0_CLIENT_ID&response_type=id_token&connection=google-oauth2&prompt=login&scope=openid%20profile&redirect_uri=https://jwt.io")!
    let callbackUrlScheme = "https"

    @IBAction func login() {
        self.webAuthSession = ASWebAuthenticationSession(url: authURL,
                                                         callbackURLScheme: callbackUrlScheme,
                                                         completionHandler: { (callback: URL?, error: Error?) in
            guard error == nil, let callback = callback else {
                return print("Error: \(String(describing: error))")
            }

            print("Success: \(callback.absoluteString)")
        })

        self.webAuthSession?.presentationContextProvider = self
        self.webAuthSession?.start()
    }
}

extension ViewController: ASWebAuthenticationPresentationContextProviding {
    func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
        return UIApplication.shared.windows.first!
    }
}
```

https://user-images.githubusercontent.com/5055789/107442061-1d00a980-6b15-11eb-8e10-9a0707966474.mov

We'll file a bug report with Apple.

**Comment by carayolthomas** [comment]
Seems to be fixed with iOS 14.5. I can't reproduce anymore on physical iPad Pro and iPad simulators.

**Comment by Widcket** [comment]
Indeed, it's been fixed on iOS 14.5.

**Comment by BhavinBhadani** [comment]
Hi @kaushikpaperboat I am doing the same with apple login but got error in apple page that suggest `Invalid request, Invalid web redirect url`. So, do we need to configure anything from our side in apple developer account and Auth0 panel?
